packages/@o3r/rules-engine/src/stores/rulesets/rulesets.selectors.ts

```diff
@@ -93,6 +101,9 @@ export const selectRuleSetLinkComponents = createSelector(
   (ruleSets) =>
     ruleSets
       .reduce((acc: Record<string, string[]>, ruleSet: Ruleset) => {
+        if (!checkValidityRange(ruleSet)) {
```

Comment:
Not sure it is the correct way to fix it.
Why not taking `selectActiveRuleSets` instead of `selectAllRulesets `?  It sounds like make more sense :S

(same below)

Reply:
It seems that linkedComponents are filtered out 
https://github.com/AmadeusITGroup/otter/pull/2401/files#diff-927d077c8a2d283c3eafef045849507ef3adebbd7f609c7f4bace09b1dbdf99bR70

Reply:
Indeed (the name was less clear than the description) but the comment remains valid:
It is always cleaner to explicitly chain the selecters than re-applying the same function everywhere.

```typescript
const selectRuleSetsToApply = createSelector(
  selectAllRulesets,
  (ruleSets) => ruleSets.filter((ruleSet: Ruleset) => {
  const validity = ruleSet.validityRange;
  if (!validity || !validity.from && !validity.to) {
    return true;
  }

  const from = validity.from && new Date(validity.from);
  const to = validity.to && new Date(validity.to);

  if (to && !isValidDate(to) || from && !isValidDate(from)) {
    return false;
  }
  const time = Date.now();
  if (to && from) {
    return from.getTime() <= time && to.getTime() >= time;
  }

  if (from) {
    return from.getTime() <= time;
  }
  return to && to.getTime() >= time;
};

export const selectActiveRuleSets = createSelector(
  selectRuleSetsToApply,
  (ruleSets) => ruleSets.filter((ruleSet: Ruleset) => (!(ruleSet.linkedComponents?.or?.length || ruleSet.linkedComponent)));

export const selectRuleSetLinkComponents = createSelector(
  selectRuleSetsToApply,
  (ruleSets) => // ....

```